docs/guides/vscode.md

```diff
@@ -155,11 +155,31 @@ The VSCode extension is based on a [language server](https://en.wikipedia.org/wi
 
 If you have environment variables that are needed by the context and the language server, you can use one of these approaches to pass variables to the language server:
 
-- Open VSCode from a terminal that has the variables set
-- Use environment variables pulled from somewhere else dynamically (e.g. a `.env` file) in your config 
+- Open VSCode from a terminal that has the variables set already. 
+  - If you have `export ENV_VAR=value` in your shell configuration file (e.g. `.zshrc` or `.bashrc`) when initializing the terminal by default, the variables will be picked up by the language server if opened from that terminal.
+- Use environment variables pulled from somewhere else dynamically (e.g. a `.env` file) in your root project directory. This will automatically be picked up by the language server through the python environment that the extension uses.
 - Set the environment variables in the python environment that the extension uses. You can find detailed instructions [here](https://code.visualstudio.com/docs/python/environments#_environment-variables)
```

Comment:
These are the same. The way we inject `.env` is the python extension and that is the docs to it. 

```suggestion
- Use environment variables pulled from somewhere else dynamically in your `config.py` for example by connecting to a secret store
- By default, a `.env` file in your root project directory will automatically be picked up by the language server through the python environment that the extension uses. For exact details on how to set the environment variables in the Python environment that the extension uses, see [here](https://code.visualstudio.com/docs/python/environments#_environment-variables)
```